Readme.md

```diff
@@ -0,0 +1,222 @@
+* `update(dt)`: Called every frame.
+* `onDelete`: Called when the entity is removed from a scene.
+
+> It is important that you do **not** override the `constructor` function, as this is used by the engine to create the entity.
+
+Within an entity script you can also access the entity's components as properties of the obejct. In the example below, we access the 'Velocity' component of the entity as `this.velocity`:
+```js
+class Player extends engine.GameObjectBase {
+  static arg_names = ["Jump Speed"];
+  static arg_types = [engine.Types.Number];
+
+  times_jumped;
+  onCreate(jump_speed) {
+    this.jump_speed = jump_speed;
+    this.times_jumped = 0;
+  }
+  update() {
+    if (engine.KeyStates.isPressed("Space")) {
+      this.Velocity.y = this.jump_speed;
+      this.times_jumped++;
+    }
+  }
+}
+```
+
+### Component Scripts
+Component scripts define custom components which can be added to entities, either programmatically or through the editor. Component scripts must extend from the [`ComponentBase`](docs/classes/ComponentBase.md) class. The `ComponentBase` class provides a number of useful functions which can be overridden:
+* `onCreate`: Called when the owner of the component is added to a scene.
+* `onDelete`: Called when this component is deleted.
+
+Additionally, components have a `owner` property available which can be used to access the entity that owns the component.  
+The component may also have a list of dependency components stored in the `dependencies` array. This is used to ensure that the component is only added to an entity if it has all of the required components. For example, the `Velocity` component has a dependency on the `Position` component.
+
+This is an example of a Position component:
+```js
+class Position extends engine.Component {
+  static arg_names = ["x", "y"];
+  static arg_types = [engine.Types.Number, engine.Types.Number];
+
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+
+  distanceTo(other) {
+    return Math.sqrt((this.x - other.x) ** 2 + (this.y - other.y) ** 2);
+  }
+}
+```
+
+### System Scripts
+System scripts define custom systems which can be added to scenes which act on a component. Systems are added to a scene automatically if an entity with the desired component is added to the scene. They can also be added manually through the [`Scene`](docs\classes\Scene.md) or [`SceneManager`](docs\classes\SceneManager.md) class.
+> Note that currently the system name must match the component name to be added automatically.
+> 
+System scripts must extend from the [`SystemBase`](docs/classes/SystemBase.md) class. The `SystemBase` class provides the `update(entities)` function which is called every frame. The `entities` parameter is an array of entities which have the component that the system is associated with. To define the component that the system is associated with, you must set the `component` property of the class to the component class.
```

Comment:
Again, this link 404s because the base class is just called `System` not `SystemBase`